rapid7_insightvm_cloud/icon_rapid7_insightvm_cloud/util/api.py

```diff
@@ -0,0 +1,91 @@
+import json
+
+from requests import request
+from logging import Logger
+
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import PluginException
+from requests.exceptions import HTTPError
+
+
+class IVM_Cloud:
+
+    _ERRORS = {
+        400: "Bad Request",
+        401: "Unauthorized",
+        500: "Internal Server Error",
+        503: "Service Unavailable",
+        000: "Unknown Status Code",
+    }
+
+    def __init__(self, token: str, logger: Logger, url: str):
+        self.logger = logger
+        self.token = token
+        self.base_url = url
+
+    def call_api_pages(self, path: str, request_type: str, pages: int, params: dict = None):
```

Comment:
I tend to think we might want to remove `call_api_pages` for the time being. There are some pretty big data sizing concerns using this and getting "all data" and would like to avoid it for the time being.